libc_impl.c

```diff
@@ -1035,10 +1038,13 @@ int wrapper_fclose(uint8_t *mem, uint32_t fp_addr) {
 static int flush_all(uint8_t *mem) {
     struct FILE_irix *f = (struct FILE_irix *)&MEM_U32(IOB_ADDR);
     int ret = 0;
-    for (int i = 0; i < NFILE; i++) {
-        if (f->_flag & IOWRT) {
+    for (int i = 0; i < g_file_max; i++) {
+        if (f[i]._flag & IOWRT) {
             ret |= wrapper_fflush(mem, IOB_ADDR + i * sizeof(struct FILE_irix));
         }
+        if (f[i]._flag & IOMYBUF) {
+            wrapper_free(mem, f[i]._base_addr);
+        }
```

Comment:
Don't have this since wrapper_fflush calls this.

Reply:
I don't see where it's called

Reply:
```C
int wrapper_fflush(uint8_t *mem, uint32_t fp_addr) {
    if (fp_addr == 0) {
        // Flush all
        return flush_all(mem);
    }
```